### PR TITLE
Name pages by state abbreviation

### PR DESCRIPTION
**Pull request by yozlet**
This changes the ways that the URLs for individual states are named. We now use the two-letter state/territory abbreviation in URLs instead of the localised state name.

For example, the URL paths for North Dakota were `/register/north-dakota/` and `/es/registrar/dakota-del-norte/`. They are now `/register/nd/` and `/es/registrar/nd/`. 

This greatly simplifies several things and makes the URLs more future-proof. The code is deployed to https://vote-gov-renaming.apps.cloud.gov/

@rogeruiz, can you review?

### CODE REVIEW

File: config/gulp/translation.js [hunk 1]
```diff
@@ -19,18 +19,12 @@ gulp.task('clean-translation', function () {
...
 
   return gulp.src('./content/register/*.md')
-    .pipe(rename(function (path) {
-      var file = spanishStateNames[path.basename].file;
-      path.basename = file;
-    }))
     .pipe(replace(/title = "(.+)"/, function (match, p1) {
       var name = p1.replace(/\s/g, '-').replace(/\./g, '').toLowerCase();
       var title = spanishStateNames[name].title;
       var file = spanishStateNames[name].file;
```
Comment (rogeruiz) [inline config/gulp/translation.js]: Are these actually necessary now? I think they were deleted in this PR. I don't think we need a mapping anymore, but it seems we still need the Translated name. Where are we storing that?

File: layouts/index.html [hunk 2]
```diff
@@ -27,11 +27,11 @@ <h1 class="usa-display">{{ $translation.homepage.header }}</h1>
             </option>
             {{ if eq $.Site.Params.language "english" }}
               {{ range where .Data.Pages.ByTitle "Section" "register" }}
-                <option value="{{ replace .Title "." "" | lower | urlize }}">{{ title .Title }}</option>
+                <option value="{{ .Params.state_abbreviation | lower }}">{{ title .Title }}</option>
```
Comment (rogeruiz) [inline layouts/index.html]: This looks to only be handling the English language. What about the `else` block below? Also where is `.Params.state_abbreviation` coming from?

File: nginx.conf [hunk 3]
```diff
@@ -59,7 +59,7 @@ http {
       # Convert form submission from front page into redirect to state page
       rewrite_log on;
       rewrite ^/states /register/$arg_userselection/? redirect;
-      rewrite ^/es/states /registrar/$arg_userselection/? redirect;
+      rewrite ^/es/states /es/registrar/$arg_userselection/? redirect;
```
Comment (rogeruiz) [inline nginx.conf]: Good catch! 🎉

**Comment by rogeruiz** [comment]
Taking a look at the deployed code, everything seems to be working. 👍 My questions above at this point are to satisfy my own curiosity and hopefully document the changes a bit better.

File: layouts/index.html (hunk 2 above)
Comment (yozlet) [inline layouts/index.html]: It's in the `else` block as well. `state_abbreviation` is coming from the `content` file.

File: config/gulp/translation.js (hunk 1 above)
Comment (yozlet) [inline config/gulp/translation.js]: You're right, we don't need `file` any more. Translated name is being stored in the generated Spanish content files.

Comment (yozlet) [inline config/gulp/translation.js]: OK, that line is removed. (Not sure why the PR hasn't updated.)

File: layouts/index.html (hunk 2 above)
Comment (rogeruiz) [inline layouts/index.html]: Alright, I think I understand. Aren't the abbreviations the same for both languages? Seems like we can clean up the key to simply be `register` so we can use it for both languages?

```
            {{ if eq $.Site.Params.language "english" }}
              {{ range where .Data.Pages.ByTitle "Section" "register" }}
                <option value="{{ .Params.state_abbreviation | lower }}">{{ title .Title }}</option>
              {{ end }}
            {{ else }}
              {{ range where .Data.Pages.ByTitle "Section" "registrar" }}
                <option value="{{ .Params.state_abbreviation | lower }}">{{ title .Title }}</option>
              {{ end }}
            {{ end }}
```

becomes 

```
              {{ range where .Data.Pages.ByTitle "Section" "register" }}
                <option value="{{ .Params.state_abbreviation | lower }}">{{ title .Title }}</option>
              {{ end }}
```

Comment (yozlet) [inline layouts/index.html]: Good point! I'll do that.

Comment (yozlet) [inline layouts/index.html]: Oh, wait - if we do that, then it'll always pick the English `.Title`, won't it?

**Comment by rogeruiz** [comment]
Approving this because I paired with @yozlet on the last few commits. and _it's golden_.

**Pull request merged**